### PR TITLE
Stop normalizing the left side of email addresses.

### PR DESCRIPTION
**Pull request by kitsuta**
Fixes https://github.com/magfest/ubersystem/issues/2392. We don't need to normalize the left side of emails, so now we just don't. We DO normalize the domain, so we now also normalize it when someone is trying to log in -- that way we can be sure we don't introduce mismatches.

### CODE REVIEW

**Comment by kitsuta** [comment]
The email_validator library we used seems to think that we should still normalize emails. From the [PyPi page](https://pypi.python.org/pypi/email_validator):
>Because you may get an email address in a variety of forms, you ought to replace it with its normalized form immediately prior to going into your database (during account creation), querying your database (during login), or sending outbound mail.
>
>The normalizations include lowercasing the domain part of the email address (domain names are case-insensitive), Unicode “NFC” normalization of the whole address (which turns characters plus combining characters into precomposed characters where possible and replaces certain Unicode characters (such as angstrom and ohm) with other equivalent code points (a-with-ring and omega, respectively)), replacement of fullwidth and halfwidth characters in the domain part, and possibly other UTS46 mappings on the domain part.

File: uber/utils.py [hunk 1]
```diff
@@ -533,14 +533,8 @@ def _clear_cache():
 
 def normalize_email(address):
```
Comment (RobRuana) [inline uber/utils.py]: If we make the changes I recommend above, I think this can go away entirely.

File: uber/site_sections/accounts.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def login(self, session, message='', original_location=None, **params):
 
         if 'email' in params:
             try:
-                account = session.get_account_by_email(params['email'])
+                account = session.get_account_by_email(normalize_email(params['email']))
```
Comment (jcooter) [inline uber/site_sections/accounts.py]: If we're normalizing .'s, there's no reason to not normalize other special characters.  Valid email addresses can contain:

- uppercase and lowercase Latin letters A to Z and a to z; 
- digits 0 to 9;
- special characters !#$%&'*+-/=?^_`{|}~;
- dot ., provided that it is not the first or last character unless quoted, and provided also that it does not appear consecutively unless quoted (e.g. John..Doe@example.com is not allowed but "John..Doe"@example.com is allowed);
- space and "(),:;<>@[\] characters are allowed with restrictions (they are only allowed inside a quoted string, as described in the paragraph below, and in addition, a backslash or double-quote must be preceded by a backslash);
- comments are allowed with parentheses at either end of the local-part; e.g. john.smith(comment)@example.com and (comment)john.smith@example.com are both equivalent to john.smith@example.com.

My suggestion for search would be to strip all special characters (unless wrapped in quotes), strip all comments, and covert the address to lowercase.

Comment (RobRuana) [inline uber/site_sections/accounts.py]: Agree. The only thing to consider is that we'd like to be able to express the normalization in SQL. So we may end up with a more aggressive normalization for queries (while maintaining the original email as provided by the attendee).

**Comment by earl7399** [comment]
Semi-related - I absolutely detest sites that don't follow RFCs, especially
this one. Please let's not be one of those sites.

--
Robert Earl <r.a.earl@gmail.com>
FAX: 1-440-919-5086

File: uber/models.py [hunk 3]
```diff
@@ -753,7 +754,7 @@ def search(self, text, *filters):
             if ':' in text:
                 target, term = text.split(':', 1)
                 if target == 'email':
-                    return attendees.icontains(Attendee.email, term.strip())
+                    return attendees.filter(Attendee.normalized_email == Attendee.normalize_email(term)).one()
```
Comment (RobRuana) [inline uber/models.py]: I might keep this as icontains, so users can still do partial matching:
```python
return attendees.icontains(Attendee.normalized_email, Attendee.normalize_email(term)).one()
```

**Pull request merged**